pkg/deployment/deployment_item.go

```diff
@@ -85,11 +137,38 @@ func NewDeploymentItem(ctx SharedContext, project *DeploymentProject, collection
 		di.renderedYamlPath = filepath.Join(di.RenderedDir, ".rendered.yml")
 	}
 
-	err = di.Project.loadVarsList(di.VarsCtx, di.Config.Vars)
+	// update context vars with project/parrent path,references and tags
+	div := NewDeploymentItemVars(di.Project, di.Config, di.dir)
+
+	// temp
+	//k := di.Tags.ListKeys()
+	// i := len(k)
+	// for j, t := range k {
+	// 	i = i - 1
+	// 	div.Tags.Set(fmt.Sprintf("tag%d", j), t) // as kluctl auto-generated tag-N)
+	// }
+
+	pe := strings.Split(di.RelToSourceItemDir, "/")
+	pr := append([]string{}, pe...) // reversed
+	for i := len(pr)/2 - 1; i >= 0; i-- {
+		opp := len(pr) - 1 - i
+		pr[i], pr[opp] = pr[opp], pr[i]
+	}
+	div.RelPath = pe
+	div.RevPath = pr
+	div.RelRoot = strings.Repeat("../", len(strings.Split(di.RelToSourceItemDir, "/")))
+
+	uov, err := uo.FromStruct(div)
 	if err != nil {
 		return nil, err
 	}
+	di.VarsCtx.Vars.MergeChild("this", uov)
```

Comment:
agree, it shall be merged into the di.Vars* exactly before the rendering of the give deployment item - the above was just "easy to add" way for the correct behaviour.

To achieve what you suggested, any implementation ideas?
- I see we could have di.VarsCollected (as an pointer) to some place of new object represented by`uov` from `pkg/deployment/deployment_item.go` ?

Anyway, I will fist update the MR with other changes before we move on this.

One more question, would it be ok, to change key `this` to `d`?

Reply:
@codablock do you mean for example pass DeploymentItemVars to this function?  https://github.com/kluctl/kluctl/blob/f55670d8b6e18e2c9196d9de6ccc38b51c4ccbdc/pkg/deployment/deployment_item.go#L161

Reply:
`render()` is a good place to add it. You would have to use something like:
```
	tmpVars := di.VarsCtx.Copy()

	div := di.getDeploymentItemProjectVars()
	uov, err := uo.FromStruct(div)
	if err != nil {
		return err
	}
	tmpVars.Vars.MergeChild("this", uov)
	
	return tmpVars.RenderDirectory(
		filepath.Join(di.Project.source.dir, di.RelToSourceItemDir),
		di.RenderedDir,
		excludePatterns,
		searchDirs,
	)
```

This might turn out to be a bit slow due to the `Copy()`, but lets optimise this later if it's really a problem.